banyand/liaison/grpc/grpc.go

```diff
@@ -334,9 +335,30 @@ func (s *Server) Write(TraceWriteServer v1.TraceService_WriteServer) error {
 }
 
 func (s *Server) Query(ctx context.Context, entityCriteria *v1.QueryRequest) (*v1.QueryResponse, error) {
-	log.Println("entityCriteria:", entityCriteria)
-
-	return &v1.QueryResponse{}, nil
+	message := bus.NewMessage(bus.MessageID(time.Now().UnixNano()), entityCriteria)
+	feat, errQuery := s.pipeline.Publish(data.TopicQueryEvent, message)
+	if errQuery != nil {
+		return nil, errQuery
+	}
+	msg, errFeat := feat.Get()
+	if errFeat != nil {
+		return nil, errFeat
+	}
+	queryMsg := msg.Data()
+
+	var arr []*v1.Entity
+	switch reflect.TypeOf(queryMsg).Kind() {
```

Comment:
You actually do not need reflect here.

Just use [type-assertion](https://tour.golang.org/methods/15), see the examples.